pkgs/by-name/op/opengamepadui/package.nix

```diff
@@ -0,0 +1,154 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  cargo,
+  fetchFromGitHub,
+  alsa-lib,
+  autoPatchelfHook,
+  dbus,
+  fontconfig,
+  gamescope,
+  gnumake,
+  godot_4,
+  godot_4-export-templates,
+  libGL,
+  libX11,
+  libXcursor,
+  libXext,
+  libXfixes,
+  libXi,
+  libXinerama,
+  libXrandr,
+  libXrender,
+  libxkbcommon,
+  openssl,
+  pkg-config,
+  vulkan-loader,
+  xorg,
+  udev,
+  libpulseaudio,
+  hwdata,
+  glxinfo,
+  upower,
+  withDebug ? false,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "opengamepadui";
+  version = "0.35.1";
+
+  buildType = "${if withDebug then "debug" else "release"}";
+
+  src = fetchFromGitHub {
+    owner = "ShadowBlip";
+    repo = "OpenGamepadUI";
+    tag = "v${version}";
+    hash = "sha256-741itXn39MxVMTNN9Azx2n5L1PRo5UsRsm6JSeY9nSI=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoVendor {
+    inherit src;
+    sourceRoot = "source/${cargoRoot}";
+    hash = "sha256-sTzMewIfKHbmVhSPZgUIzFFz1ahK+PMoQ5oB4GEt8nY=";
+  };
+  cargoRoot = "extensions";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cargo
+    dbus.dev
+    gnumake
+    godot_4
+    godot_4-export-templates
+    openssl
+    pkg-config
+    rustPlatform.cargoSetupHook
+    xorg.libX11.dev
+    xorg.libXi.dev
+    xorg.libXres.dev
+    xorg.libXtst
+  ];
+
+  buildInputs = [
+    dbus
+    dbus.lib
+    fontconfig
+    fontconfig.lib
+    stdenv.cc.cc.lib
+  ];
+
+  runtimeDependencies = [
+    alsa-lib
+    dbus
+    dbus.lib
+    gamescope
+    fontconfig
+    fontconfig.lib
+    glxinfo
+    hwdata
+    libGL
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+    libpulseaudio
+    libxkbcommon
+    udev
+    upower
+    vulkan-loader
+    xorg.libX11
+    xorg.libXi
+    xorg.libXres
+    xorg.libXtst
+  ];
+
+  dontStrip = withDebug;
+
+  buildPhase = ''
+    runHook preBuild
+
+    export GODOT="${godot_4}/bin/godot4"
+    export GODOT_VERSION=$($GODOT --version | grep -o '[0-9].*[0-9]\.' | sed 's/\.$//')
+    export GODOT_RELEASE=$($GODOT --version | grep -oP '^[0-9].*?[a-z]\.' | grep -oP '[a-z]+')
+
+    # Godot looks for export templates in HOME
+    export HOME=$(mktemp -d)
+    mkdir -p $HOME/.local/share/godot/export_templates
+    ln -s "${godot_4-export-templates}" "$HOME/.local/share/godot/export_templates/$GODOT_VERSION.$GODOT_RELEASE"
+
+    export BUILD_TYPE=${buildType}
+    make clean
+    make build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    export EXPORT_TEMPLATE="${godot_4-export-templates}"
+    export BUILD_TYPE=${buildType}
+    make GODOT="${godot_4}/bin/godot4" install PREFIX=$out
+
+    # The Godot binary looks in "../lib" for gdextensions
+    mkdir -p $out/share/lib
+    mv $out/share/opengamepadui/*.so $out/share/lib
+
+    runHook postInstall
+  '';
+
```

Comment:
`passthru.updateScript = gitUpdater { rev-prefix = "v"; };`

Reply:
Are you sure the gitUpdater will take care of `cargoDeps`?

Reply:
I tested this with another package. Not actually. For the cargo case it will be left behind.

Reply:
Why not use `nix-update-script`, which seems to handle just about everything?

Reply:
Let's use it then

Reply:
I think there were no update script then I suggested one.